src/test/java/com/cloudbees/jenkins/GitHubPushHookReceiverTest.java

```diff
@@ -0,0 +1,103 @@
+package com.cloudbees.jenkins;
+
+import hudson.model.Result;
+import hudson.model.FreeStyleProject;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.plugins.git.SubmoduleConfig;
+import hudson.plugins.git.BranchSpec;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.UserRemoteConfig;
+import hudson.plugins.git.util.Build;
+import hudson.plugins.git.util.BuildData;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.concurrent.ExecutionException;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/**
+ * Test case covering the code responsible to receive the push notification
+ * from GitHub.
+ */
+public class GitHubPushHookReceiverTest extends Assert {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Inject
+    GitHubWebHook receiver;
+
+    @Test
+    public void receivePushHookOnFreeStyle() throws IOException, InterruptedException, ExecutionException {
+        j.jenkins.getInjector().injectMembers(this);
```

Comment:
@amuniz yep, its the same, but more powerful
https://github.com/junit-team/junit/wiki/Rules